src/ChannelAdam.Soap/Internal/Soap11BodyBuilder.cs

```diff
@@ -30,14 +30,16 @@ public class Soap11BodyBuilder : ISoap11BodyBuilder
 
         private XElement? bodyElement;
         private readonly ISoap11EnvelopeBuilder envelopeBuilder;
+        private readonly string? prefix;
 
         #endregion Private Fields
 
         #region Internal Constructors
 
-        internal Soap11BodyBuilder(ISoap11EnvelopeBuilder envelopeBuilder)
+        internal Soap11BodyBuilder(ISoap11EnvelopeBuilder envelopeBuilder, string? prefix)
```

Comment:
Can you please rename this parameter to envelopeNamespacePrefix ?  (perhaps in all the parameters)
Especially in circumstances like this, where the Body is the topic of discussion, it is ambiguous or at least unclear what "prefix" means.
Also, I was wondering whether it made sense to expose this prefix on the ISoap11EnveloperBuilder as a property - and that way this extra parameter is not required here.

Reply:
Right the name is ambiguous will make the change to envelopeNamespacePrefix.

Reply:
Second by exposing the prefix in ISoap11EnvelopeBuilder it would be a benefit in not having extra parameter in the SoapBodyBuilder. The reason why I initially did not add it as  a property is because I thought it would clutter the ISoap11EnvelopeBuilder. 

But I see no reason to not add it 👍

Reply:
Instead maybe add a SetNamespacePrefix in ISoap11BodyBuilder? 

Kind of how it is done in [Java](https://docs.oracle.com/javase/8/docs/api/javax/xml/soap/SOAPBody.html)? 

```java
      String prefix = "soap";
      SOAPMessage soapMessage = soapResponse.getSaajMessage();
      SOAPPart soapPart = soapMessage.getSOAPPart();
      SOAPEnvelope envelope = soapPart.getEnvelope();
      SOAPHeader header = soapMessage.getSOAPHeader();
      SOAPBody body = soapMessage.getSOAPBody();
      SOAPFault fault = body.getFault();
      envelope.setPrefix(prefix);
      header.setPrefix(prefix);
      body.setPrefix(prefix);
```